FluentNest/Statistics.cs

```diff
@@ -139,29 +140,44 @@ public static class Statistics
         public static AggregationContainerDescriptor<T> DistinctBy<T>(this AggregationContainerDescriptor<T> agg, Expression<Func<T, object>> fieldGetter) where T : class
         {
             var aggName = fieldGetter.GetAggName(AggType.Distinct);
+            var namedField = Names.GetNameFromGetFieldNamed(fieldGetter.Body);
+
+            if (namedField != null)
+            {
+                return agg.Terms(aggName, x => x.Field(namedField).Size(int.MaxValue));
+            }
+
             return agg.Terms(aggName, x => x.Field(fieldGetter).Size(int.MaxValue));
         }
 
-        public static AggregationContainerDescriptor<T> TopHits<T>(this AggregationContainerDescriptor<T> agg, int size, params Expression<Func<T, object>>[] fieldGetter) where T : class
+        public static AggregationContainerDescriptor<T> TopHits<T>(this AggregationContainerDescriptor<T> agg, int size, params Expression<Func<T, object>>[] fieldGetters) where T : class
         {
             var aggName = AggType.TopHits.ToString();
-            return agg.TopHits(aggName, x => x.Size(size).Source(i=>i.Includes(f=>f.Fields(fieldGetter))));
+
+            var namedFields = fieldGetters.Select(x => Names.GetNameFromGetFieldNamed(x.Body)).Where(x => !string.IsNullOrEmpty(x)).ToArray();
+            var fieldGettersWithoutNamedFields = fieldGetters.Where(x => Names.GetNameFromGetFieldNamed(x.Body) == null).ToArray();

```

Comment:
Method might not be cheap in the degenerate case :)

Reply:
Indeed, I think it's unlikely it will be abused that way though so it should be ok. Tbh I was not happy myself, I tried other alternatives but I was happier with the explicitness of this one in the end.